modules/candidate_parameters/jsx/probandInfo.js

```diff
@@ -103,9 +103,40 @@ var ProbandInfo = React.createClass(
       if (this.state.formData.ProbandGender !== null) {
         dobRequired = true;
       }
-      if (this.state.formData.ProbandDoB !== null) {
-        dob2Required = true;
-      }
+        var dob = this.state.formData.ProbandDoB;
+
+        if (dob !== null && dob !== undefined) {
+            dob2Required = true;
+        }
+
+        var ageDifference = this.state.Data.ageDifference;
+        if (dob !== null && dob !== undefined) {
+
+           var candidateDOB = this.state.Data.candidateDOB;
```

Comment:
Agreed, and if you are going to create a function for this you might as well make it a [pure function](https://en.wikipedia.org/wiki/Pure_function) so that it is easily testable.